src/logging.c

```diff
@@ -86,6 +88,37 @@ rcutils_ret_t rcutils_logging_initialize(void)
   return rcutils_logging_initialize_with_allocator(rcutils_get_default_allocator());
 }
 
+// A utility function to get zero or one from an environment variable.  Returns
+// -2 if we failed to get the environment variable, -1 if the value in the
+// environment variable is something other than "0" or "1", 0 if the environment
+// variable is "0" or "", or "1" if the environment variable is "1".
+static int rcutils_get_env_var_zero_or_one(
+  const char * name, const char * zero_semantic,
+  const char * one_semantic)
+{
+  const char * env_var_value;
+  const char * ret_str = rcutils_get_env(name, &env_var_value);
+  if (ret_str != NULL) {
+    fprintf(
+      stderr, "Error getting env. variable "
+      "%s: %s\n", name, ret_str);
+    return -2;
+  }
+
+  if (strcmp(env_var_value, "0") == 0 || strcmp(env_var_value, "") == 0) {
+    return 0;
+  } else if (strcmp(env_var_value, "1") == 0) {
+    return 1;
+  }
+
+  fprintf(
+    stderr,
+    "Warning: unexpected value [%s] specified for %s. "
+    "Valid values are 0 (%s) or 1 (%s).  Default value 0 will be used.\n",
+    env_var_value, name, zero_semantic, one_semantic);
+  return -1;
```

Comment:
@clalancette why not returning `0` then?

Reply:
That's an excellent point, and simplifies the return structure from this function.  Will do that, thanks for the idea.